### ISSUE TITLE
StringIndexOutOfBoundsException while editing code with here documents

### ISSUE THREAD

**Issue by serbani** [de-jcup/eclipse-bash-editor#106]
Opening a script containing the following snippet results in an exception.
```
cat <<-" OF"
```

Initially the exception was thrown while editing the code and deleting some here documents syntax, but I managed to save the file at that point. Reopening it resulted in an "Editor could not be initialized."

```
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:1967)
	at de.jcup.basheditor.script.parser.HereDocParserSupport.isLiteralWhenFirstAndLastCharsRemoved(HereDocParserSupport.java:204)
	at de.jcup.basheditor.script.parser.HereDocParserSupport.isEndLiteralFound(HereDocParserSupport.java:197)
	at de.jcup.basheditor.script.parser.HereDocParserSupport.step2_scanForContent(HereDocParserSupport.java:67)
	at de.jcup.basheditor.script.parser.HereDocParserSupport.isHereDocStateHandled(HereDocParserSupport.java:18)
	at de.jcup.basheditor.document.HereDocumentRule.evaluate(HereDocumentRule.java:41)
	at de.jcup.basheditor.document.HereDocumentRule.evaluate(HereDocumentRule.java:23)
	at org.eclipse.jface.text.rules.RuleBasedScanner.nextToken(RuleBasedScanner.java:162)
	at org.eclipse.jface.text.rules.RuleBasedPartitionScanner.nextToken(RuleBasedPartitionScanner.java:87)
	at org.eclipse.jface.text.rules.FastPartitioner.initialize(FastPartitioner.java:173)
	at org.eclipse.jface.text.rules.FastPartitioner.checkInitialization(FastPartitioner.java:148)
	at org.eclipse.jface.text.rules.FastPartitioner.connect(FastPartitioner.java:140)
	at org.eclipse.jface.text.rules.FastPartitioner.connect(FastPartitioner.java:121)
	at de.jcup.basheditor.document.BashFileDocumentProvider.createDocument(BashFileDocumentProvider.java:37)
	at org.eclipse.ui.editors.text.FileDocumentProvider.createElementInfo(FileDocumentProvider.java:716)
	at org.eclipse.ui.texteditor.AbstractDocumentProvider.connect(AbstractDocumentProvider.java:392)
	at org.eclipse.ui.texteditor.AbstractTextEditor.doSetInput(AbstractTextEditor.java:4178)
	at org.eclipse.ui.texteditor.StatusTextEditor.doSetInput(StatusTextEditor.java:257)
	at org.eclipse.ui.texteditor.AbstractDecoratedTextEditor.doSetInput(AbstractDecoratedTextEditor.java:1466)
	at org.eclipse.ui.editors.text.TextEditor.doSetInput(TextEditor.java:150)
	at de.jcup.basheditor.BashEditor.doSetInput(BashEditor.java:387)
	at org.eclipse.ui.texteditor.AbstractTextEditor$5.run(AbstractTextEditor.java:3154)
	at org.eclipse.jface.operation.ModalContext.runInCurrentThread(ModalContext.java:437)
	at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:353)
	at org.eclipse.ui.internal.WorkbenchWindow$14.run(WorkbenchWindow.java:2195)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:70)
	at org.eclipse.ui.internal.WorkbenchWindow.run(WorkbenchWindow.java:2191)
	at org.eclipse.ui.texteditor.AbstractTextEditor.internalInit(AbstractTextEditor.java:3172)
	at org.eclipse.ui.texteditor.AbstractTextEditor.init(AbstractTextEditor.java:3197)
	at de.jcup.basheditor.BashEditor.init(BashEditor.java:518)
	at org.eclipse.ui.internal.EditorReference.initialize(EditorReference.java:362)
	at org.eclipse.ui.internal.e4.compatibility.CompatibilityPart.create(CompatibilityPart.java:333)
	at sun.reflect.GeneratedMethodAccessor353.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
...
```

Thank you for this nice plugin!

**Comment by de-jcup** [comment]
Problems with rebasing occured must apply bugfix again...

**Comment by de-jcup** [comment]
@serbani I just released version 1.5.3 to eclipse marketplace, it contains the bugfix.

**Comment by serbani** [comment]
Thank you for the quick fix, works like a charm.
